src/main.cpp

```diff
@@ -433,30 +329,168 @@ BOOL __stdcall AutoDetectCallback (GUID*    lpGUID,
     return TRUE;
 }
 
+
+MM2RawFnHook<WNDPROC> gfxWindowProc(NULL, NULL, 0x04A88F0);
+
 class gfxHandler
 {
 public:
-    static bool gfxAutoDetect(bool *success)
+    static void setRes(int width, int height, int cdepth, int zdepth, bool detectArgs)
+    {
+        LogFile::WriteLine("Additional graphics params enabled.");
+
+        $setRes(width, height, cdepth, zdepth, datArgParser::Get("gfxArgs") ? true : detectArgs);
```

Comment:
Oh hey, sorry about the "gfxArgs" thing. I was trying to debug why I had some strange flickering issues. This can be removed.